### DISCUSSION TITLE
How to reference the current file path (e.g. for use in shell commands)?

### DISCUSSION THREAD

**Discussion by candiedyams** [helix-editor/helix#3256] [Q&A]
If I just want to run the current (executable) file, I'd like to run `:sh <current_file>`. But I wasn't seeing in the [commands](https://docs.helix-editor.com/commands.html) or [keymap](https://docs.helix-editor.com/keymap.html) how to reference the file path for the current buffer. How best can I do that?

**Comment by kirawi** [comment]
There currently isn't a way, but #3008 will allow you to use a `%` register to paste the filename.
